udf-compiler/src/main/scala/com/nvidia/spark/udf/Instruction.scala

```diff
@@ -323,6 +349,17 @@ case class Instruction(opcode: Int, operand: Int, instructionStr: String) extend
     State(locals.updated(localsIndex, top), rest, cond, expr)
   }
 
+  private def athrow(state: State): State = {
+    val State(locals, top :: rest, cond, expr) = state
+    if (!top.isInstanceOf[Repr.SparkExcept]) {
+      throw new SparkException("Unsupported type for athrow")
+    }
+    // Empty the stack and convert the internal representation of
+    // org.apache.spark.SparkException object to RaiseError, then push it to the
+    // stack.
+    State(locals, List(RaiseError(top.asInstanceOf[Repr.SparkExcept].message)), cond, expr)
```

Comment:
I believe one approach here would be to code up a new catalyst node `RaiseErrorV2` or `RaiseSpecificError` or something like this, where it has the CPU `eval` and `doGenCode`, as it would be very much like `RaiseError`, but it has an extra parameter which is the specific type of exception to throw. It would then instantiate this exception and throw the specific type.

For the gpu, we could replace this new node with the GPU specific type, and for the CPU it would work the way the caller intended. @jlowe @revans2 for some :eyes:

Reply:
Updated `udf-to-catalyst-expressions.md`